app/experimenter/experiments/models/nimbus.py

```diff
@@ -124,31 +122,53 @@ class Meta:
         verbose_name = "Nimbus Experiment"
         verbose_name_plural = "Nimbus Experiments"
 
+    class Filters:
+        IS_LAUNCH_QUEUED = Q(
+            status=NimbusConstants.Status.DRAFT,
+            publish_status=NimbusConstants.PublishStatus.APPROVED,
+        )
+        IS_LAUNCHING = Q(
+            status=NimbusConstants.Status.DRAFT,
+            publish_status=NimbusConstants.PublishStatus.WAITING,
+        )
+        IS_PAUSE_QUEUED = Q(
+            status=NimbusConstants.Status.LIVE,
+            is_paused=False,
+        )
+        IS_END_QUEUED = Q(
+            status=NimbusConstants.Status.LIVE,
+            publish_status=NimbusConstants.PublishStatus.APPROVED,
+            is_end_requested=True,
+        )
+        IS_ENDING = Q(
+            status=NimbusConstants.Status.LIVE,
+            publish_status=NimbusConstants.PublishStatus.WAITING,
+            is_end_requested=True,
+        )
+        SHOULD_TIMEOUT = Q(IS_LAUNCHING | IS_ENDING)
```

Comment:
Love it, so you can just infinitely nest `Q` calls?

Reply:
Yeeeee they're composable, makes declaring stuff like this very nice and succinct, like algebraic types in TS or Rust or any ML variant.